zipline/data/bundles/core.py

```diff
@@ -0,0 +1,467 @@
+
+        See Also
+        --------
+        zipline.data.bundles.bundles
+        """
+        if name in bundles:
+            warnings.warn(
+                'Overwriting bundle with name %r' % name,
+                stacklevel=3,
+            )
+        _bundles[name] = _BundlePayload(
+            calendar,
+            opens,
+            closes,
+            minutes_per_day,
+            f,
+        )
+        return f
+
+    def unregister(name):
+        """Unregister a bundle.
+
+        Parameters
+        ----------
+        name : str
+            The name of the bundle to unregister.
+
+        Raises
+        ------
+        UnknownBundle
+            Raised when no bundle has been registered with the given name.
+
+        See Also
+        --------
+        zipline.data.bundles.bundles
+        """
+        try:
+            del _bundles[name]
+        except KeyError:
+            raise UnknownBundle(name)
+
+    def ingest(name,
+               environ=os.environ,
+               timestamp=None,
+               show_progress=True):
+        """Ingest data for a given bundle.
+
+        Parameters
+        ----------
+        name : str
+            The name of the bundle.
+        environ : mapping, optional
+            The environment variables. By default this is os.environ.
+        timestamp : datetime, optional
+            The timestamp to use for the load.
+            By default this is the current time.
+        show_progress : bool, optional
+            Tell the ingest function to display the progress where possible.
+        """
+        try:
+            bundle = bundles[name]
+        except KeyError:
+            raise UnknownBundle(name)
+
+        if timestamp is None:
+            timestamp = pd.Timestamp.utcnow()
+        timestamp = timestamp.tz_convert('utc').tz_localize(None)
+        timestr = timestamp.isoformat()
+        cachepath = cache_path(name, timestr, environ=environ)
+        pth.ensure_directory(cachepath)
+
+        with dataframe_cache(cachepath, clean_on_failure=False) as cache, \
+                transactional_dir(
+                    daily_equity_path(name, timestr, environ=environ),
+                ) as daily_bars_dir, \
+                transactional_dir(
+                    minute_equity_path(name, timestr, environ=environ),
+                ) as minute_bars_dir, \
+                transactional_file(
+                    asset_db_path(name, timestr, environ=environ),
+                ) as asset_db_file, \
+                transactional_file(
+                    adjustment_db_path(name, timestr, environ=environ),
+                ) as adjustment_db_file:
+            # we use `cleanup_on_failure=False` so that we don't purge the
+            # cache directory if the load fails in the middle
+            daily_bar_writer = BcolzDailyBarWriter(
+                daily_bars_dir.name,
+                bundle.calendar,
+            )
+            daily_bar_writer.write(())
```

Comment:
to ensure that the ctables exist when we try to create the adjustment reader which will eagerly open the tables.